RecoHGCal/TICL/plugins/TracksterLinkingbyFastJet.h

```diff
@@ -0,0 +1,60 @@
+#ifndef RecoHGCal_TICL_TracksterLinkingAlgoByFastJet_H
+#define RecoHGCal_TICL_TracksterLinkingAlgoByFastJet_H
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "RecoHGCal/TICL/interface/TracksterLinkingAlgoBase.h"
+#include "fastjet/ClusterSequence.hh"
+#include "DataFormats/Math/interface/deltaR.h"
+
+namespace ticl {
+
+  class TracksterLinkingbyFastJet : public TracksterLinkingAlgoBase {
+  public:
+    TracksterLinkingbyFastJet(const edm::ParameterSet& conf, edm::ConsumesCollector iC)
+        : TracksterLinkingAlgoBase(conf, iC), radius_(conf.getParameter<double>("radius")) {
+      // Cluster tracksters into jets using FastJet with configurable algorithm
+      auto algo = conf.getParameter<int>("jet_algorithm");
+
+      switch (algo) {
+        case 0:
+          algorithm_ = fastjet::kt_algorithm;
+          break;
+        case 1:
+          algorithm_ = fastjet::cambridge_algorithm;
+          break;
+        case 2:
+          algorithm_ = fastjet::antikt_algorithm;
+          break;
+      }
```

Comment:
Should this throw an exception if the `algo` is something else?

Reply:
yes, added